proposals/lip-0054.md

```diff
@@ -665,6 +675,13 @@ def verify(trs: Transaction) -> None:
     if sidechainAccount.status != CHAIN_STATUS_TERMINATED
```

Comment:
Unrelated to this PR, but aren't we allowing to terminate sidechains which are in the registered status and did not activate yet?

Reply:
On line 673, we should also validate the sidechainAccount object against the schema

Reply:
> Unrelated to this PR, but aren't we allowing to terminate sidechains which are in the registered status and did not activate yet?

Yes, we do. In this case, the status is `CHAIN_STATUS_TERMINATED`. Hence, the condition is not fulfilled such that NO exception is raced, right?

> On line 673, we should also validate the sidechainAccount object against the schema, similar to what we do for message recovery initialization: validateObjectSchema(channelDataSchema, trsParams.channel)

I added the schema validation. I also added the decoding step to message recovery initialization which was missing there.

Reply:
>But i think we do not want to allow to terminate registered sidechains, as the liveness requirement should not be enforced yet.

I'm not 100%ly sure what you mean. Do you think there is something wrong in our rules how/when to terminate a sidechain, or in the verification of the State Recovery Initialization Command?

Reply:
I fixed the check for the chain status